docs/docs/getting-started/bare-metal.md

```diff
@@ -28,7 +28,70 @@ Follow Canonical's instructions on [setting up Intel TDX on Ubuntu 24.04](https:
 
 Increase the `user.max_inotify_instances` sysctl limit by adding `user.max_inotify_instances=8192` to `/etc/sysctl.d/99-sysctl.conf` and running `sysctl --system`.
 
-## K3s Setup
+## K3s setup
 
 1. Follow the [K3s setup instructions](https://docs.k3s.io/) to create a cluster.
 2. Install a block storage provider such as [Longhorn](https://docs.k3s.io/storage#setting-up-longhorn) and mark it as the default storage class.
+
+## Preparing a cluster for GPU usage
+
+:::warning
+Currently, Contrast only supports GPU workloads on SEV-SNP-based clusters.
+:::
+
+To enable GPU usage on a Contrast cluster, some conditions need to be fulfilled for *each cluster node* that should host GPU workloads:
+
+1. The IOMMU needs to be activated. This can be checked by running `ls /sys/kernel/iommu_groups`. If the output contains the group indices (`0`, `1`, ...), the IOMMU is supported on the host.
+   If it doesn't, `intel_iommu=on` needs to be added to the kernel command line.
+2. Additionally, the host kernel needs to have the following kernel configuration options enabled:
+    - `CONFIG_VFIO`
+    - `CONFIG_VFIO_IOMMU_TYPE1`
+    - `CONFIG_VFIO_MDEV`
+    - `CONFIG_VFIO_MDEV_DEVICE`
+    - `CONFIG_VFIO_PCI`
+
+If the per-node requirements are fulfilled, the NVIDIA GPU Operator needs to be deployed to the cluster. It manages the facilitation of pod-VMs with GPUs via VFIO.
+
+Initially, label all nodes that *should run GPU workloads*:
+
+```sh
+kubectl label node <node-name> nvidia.com/gpu.workload.config=vm-passthrough
+```
+
+For a general setup, the operator can then be deployed like so:
+
+```sh
+helm install --wait --generate-name \
+   -n gpu-operator --create-namespace \
+   nvidia/gpu-operator \
+   --version=v24.9.1 \
+   --set sandboxWorkloads.enabled=true \
+   --set sandboxWorkloads.defaultWorkload='vm-passthrough' \
+   --set nfd.nodefeaturerules=true \
+   --set vfioManager.enabled=true \
+   --set kataManager.enabled=true \
+   --set ccManager.enabled=true
+```
+
+These settings might be tweaked specific to the use-case. For example, `ccManager.enabled` (the confidential computing manager daemonset) is superfluous if no CC-capable GPUs like the NVIDIA H100 are available for the given deployment.
+The [official installation instructions](https://docs.nvidia.com/datacenter/cloud-native/gpu-operator/latest/getting-started.html) can be consulted as a guideline of what options are required for the specific use-case.
+For any GPU-enabled Contrast deployment, the minimal set of required options is the aforementioned one, excluding `ccManager.enabled`.
```

Comment:
I'm not sure if we should just keep it simple and ignore the case of connecting non CC-GPUs to Contrast.
Might be confusing?

Reply:
I think 99% of people don't have access to H100 GPUs, even in "professional" setups. What do you think @thomasten?

Reply:
If I understand correctly, the command above also works for non CC-GPUs, it just unnecessarily deploys the CC manager. I'd then shorten this section to something like "See the official installation instructions for details and further options."